table.go

```diff
@@ -196,6 +196,34 @@ func (rt *RoutingTable) peersToValidate() []PeerInfo {
 	return rt.peersForValidationFnc(peers)
 }
 
+// NPeersForCPL returns the number of peers we have for a given Cpl
+func (rt *RoutingTable) NPeersForCpl(cpl uint) int {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+
+	// it's in the last bucket
+	if int(cpl) >= len(rt.buckets)-1 {
+		count := 0
+		b := rt.buckets[len(rt.buckets)-1]
+		for _, p := range b.peerIds() {
+			if CommonPrefixLen(rt.local, ConvertPeerID(p)) == int(cpl) {
```

Comment:
Ouch. We should probably start storing these converted IDs. Sha256 is going to kill us.

Reply:
@Stebalien Great point. I never thought about the cost we pay every time we call this function. I've created an issue here:

https://github.com/libp2p/go-libp2p/issues/812